tools/pd-api-bench/main.go

```diff
@@ -39,18 +46,40 @@ import (
 )
 
 var (
-	qps   = flag.Int64("qps", 1000, "qps")
-	burst = flag.Int64("burst", 1, "burst")
-
-	httpCases = flag.String("http-cases", "", "http api cases")
-	gRPCCases = flag.String("grpc-cases", "", "grpc cases")
+	qps, burst           int64
+	httpCases, gRPCCases string
 )
 
-var base = int64(time.Second) / int64(time.Microsecond)
+var (
+	pdAPIExecutionHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "pd",
+			Subsystem: "api_bench",
+			Name:      "pd_api_execution_duration_seconds",
+			Help:      "Bucketed histogram of all pd api execution time (s)",
+			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20), // 1ms ~ 524s
+		}, []string{"type"})
+
+	pdAPIRequestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "pd",
+			Subsystem: "api_bench",
+			Name:      "pd_api_request_total",
+			Help:      "Counter of the pd http api requests",
+		}, []string{"type", "result"})
+)
 
 func main() {
+	prometheus.MustRegister(pdAPIExecutionHistogram)
+	prometheus.MustRegister(pdAPIRequestCounter)
+
+	ctx, cancel := context.WithCancel(context.Background())
 	flagSet := flag.NewFlagSet("api-bench", flag.ContinueOnError)
 	flagSet.ParseErrorsWhitelist.UnknownFlags = true
+	flagSet.Int64Var(&qps, "qps", 1, "qps")
+	flagSet.Int64Var(&burst, "burst", 1, "burst")
+	flagSet.StringVar(&httpCases, "http-cases", "", "http api cases")
+	flagSet.StringVar(&gRPCCases, "grpc-cases", "", "grpc cases")
```

Comment:
Do we need them?

Reply:
This is to keep the original command line arguments, which is more convenient for simple testing